obspy/core/utcdatetime.py

```diff
@@ -383,8 +388,29 @@ def __init__(self, *args, **kwargs):
             kwargs['second'] = int(_sec)
             kwargs['microsecond'] = int(round(_frac * 1e6))
             args = args[0:5]
-        dt = datetime.datetime(*args, **kwargs)
-        self._from_datetime(dt)
+
+        try:  # If a value Error is raised try to allow overflow (see #2222)
+            dt = datetime.datetime(*args, **kwargs)
+        except ValueError:
+            if args:
+                raise
+            else:
+                # ensure hour minute second are in kwargs
+                for time_unit in ('hour', 'minute', 'second'):
+                    kwargs[time_unit] = kwargs.get(time_unit, 0)
+                extra_seconds = 0
+                # keep track of second equiv. of overflow
+                extra_seconds += (kwargs['hour'] // 24) * 3600 * 24
+                extra_seconds += (kwargs['minute'] // 60) * 60 * 60
+                extra_seconds += (kwargs['second'] // 60) * 60
+                # reduce value in kwargs to be within normal bounds
+                kwargs['hour'] %= 24
+                kwargs['minute'] %= 60
+                kwargs['second'] %= 60
+            # get the correct timestamp, add extra seconds
+            self._ns = (UTCDateTime(**kwargs) + extra_seconds).ns
```

Comment:
Do we really need the extra object? Set `dt` here and `extra_seconds = 0` in the alternative case.

Reply:
2 things here:

1. With the try/except configuration as it is now adding this new behavior will not induce a performance hit when the normal dt construction works. Since `UTCDateTime` is used extensively I would be reluctant to add overhead to its `__init__` in all cases.

2. We need to avoid changing the `ns` property once it is set, else a depreciation warning will be issued by [this method](https://github.com/obspy/obspy/blob/befecf33ab2a1e2b5ceb4f5b0617cc37e055822b/obspy/core/utcdatetime.py#L1171-L1179. )